test/shopify-cli/admin_api/populate_resource_command_test.rb

```diff
@@ -22,6 +22,17 @@ def test_with_schema_args_overrides_input
         assert_equal('[{price: "4.99"}]', @resource.input['variants'])
       end
 
+      def test_prompts_and_writes_to_env_if_no_shop
+        Project.current.stubs(:env).returns(Resources::EnvFile.new(api_key: '123', secret: 'foo'))
+        assert_nil(Project.current.env.shop)
```

Comment:
Not sure if this will be set with the stubbing, but if it is we could also test that it's set after the `call`.

Reply:
It wouldn't be possible to test if it's set after the call, because every use of an .env file gets stubbed with only `api_key` and `secret`. Here, it's being double-checked that whatever stubbed .env file is being used to test, is indeed missing the shop.